crates/solvers/src/tests/dex/partial_fill.rs

```diff
@@ -0,0 +1,212 @@
+//! Tests that dex solvers consecutively decrease the amounts they try to fill
+//! partially fillable orders with across `/solve` requests to eventually find a
+//! fillable amount that works.
+
+use {
+    crate::tests::{self, balancer, mock},
+    serde_json::json,
+};
+
+#[tokio::test]
+async fn test() {
+    let inner_request = |amount| {
+        json!({
+            "sellToken": "0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
+            "buyToken": "0xba100000625a3754423978a60c9317c58a424e3d",
+            "orderKind": "sell",
+            "amount": amount,
+            "gasPrice": "15000000000",
+        })
+    };
+
+    let no_swap_found_response = json!({
+        "tokenAddresses": [],
+        "swaps": [],
+        "swapAmount": "0",
+        "swapAmountForSwaps": "0",
+        "returnAmount": "0",
+        "returnAmountFromSwaps": "0",
+        "returnAmountConsideringFees": "0",
+        "tokenIn": "0x0000000000000000000000000000000000000000",
+        "tokenOut": "0x0000000000000000000000000000000000000000",
+        "marketSp": "0",
+    });
+
+    let api = mock::http::setup(vec![mock::http::Expectation::Post {
```

Comment:
Formatting looks weird here for some reason, maybe there is a line that is too long somewhere 👇.